packages/eightballer/connections/websocket_client/connection.py

```diff
@@ -0,0 +1,130 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Scaffold connection and channel."""
+from typing import Any, Optional
+from aea.configurations.base import PublicId
+from aea.connections.base import BaseSyncConnection, Connection, ConnectionStates
+from aea.mail.base import Envelope
+import websocket
+
+from packages.fetchai.protocols.default.message import DefaultMessage
+
+
+CONNECTION_ID = PublicId.from_str("eightballer/websocket_client:0.1.0")
+
+from threading import Thread
+
+
+class WebSocketClient(Connection):
+    """Proxy to the functionality of the SDK or API."""
+
+    connection_id = CONNECTION_ID
+    _new_messages: list
+    _endpoint: str
+    _thread: Thread
+    _wss: websocket.WebSocket
+
+    def __init__(self, **kwargs: Any) -> None:
+        """
+        Initialize the connection.
+
+        The configuration must be specified if and only if the following
+        parameters are None: connection_id, excluded_protocols or restricted_to_protocols.
+
+        Possible keyword arguments:
+        - configuration: the connection configuration.
+        - data_dir: directory where to put local files.
+        - identity: the identity object held by the agent.
+        - crypto_store: the crypto store for encrypted communication.
+        - restricted_to_protocols: the set of protocols ids of the only supported protocols for this connection.
+        - excluded_protocols: the set of protocols ids that we want to exclude for this connection.
+
+        :param kwargs: keyword arguments passed to component base
+        """
+        self._new_messages = []
+        self._endpoint = kwargs['configuration'].config['endpoint']
+        self._target_skill_id = kwargs['configuration'].config['target_skill_id']
+        assert self._endpoint is not None, "Endpoint must be provided!"
+        super().__init__(**kwargs)  # pragma: no cover
+
+    async def connect(self) -> None:
+        """
+        Set up the connection.
+
+        In the implementation, remember to update 'connection_status' accordingly.
+        """
+        self._wss = websocket.create_connection(self._endpoint)
+        self.state = ConnectionStates.connected
+        self.logger.info("Websocket connection established.")
+        self._thread = Thread(target=self._run)
+        self._thread.start()
+
+    async def disconnect(self) -> None:
+        """
+        Tear down the connection.
+
+        In the implementation, remember to update 'connection_status' accordingly.
+        """
+        self.logger.debug("Disconnecting...")  # pragma: no cover
+        self.state = ConnectionStates.disconnected
```

Comment:
need to stop the thread here as well to actually disconnect

Reply:
cool thanks